pkgs/development/python-modules/virtualenv/0001-Check-base_prefix-and-base_exec_prefix-for-Python-2.patch

```diff
@@ -0,0 +1,37 @@
+From 21563405d6e2348ee457187f7fb61beb102bb367 Mon Sep 17 00:00:00 2001
+From: Frederik Rietdijk <fridh@fridh.nl>
+Date: Sun, 24 May 2020 09:33:13 +0200
+Subject: [PATCH] Check base_prefix and base_exec_prefix for Python 2
+
+This is a Nixpkgs-specific change so it can support virtualenvs from Nix envs.
+---
+ src/virtualenv/discovery/py_info.py | 8 ++++++--
+ 1 file changed, 6 insertions(+), 2 deletions(-)
+
+diff --git a/src/virtualenv/discovery/py_info.py b/src/virtualenv/discovery/py_info.py
+index 6f12128..74e9218 100644
+--- a/src/virtualenv/discovery/py_info.py
++++ b/src/virtualenv/discovery/py_info.py
+@@ -51,13 +51,17 @@ class PythonInfo(object):
+         self.version = u(sys.version)
+         self.os = u(os.name)
+ 
++        config_vars = {} if sys.version_info.major is not 2 else sysconfig._CONFIG_VARS
++        base_prefix = config_vars.get("prefix")
++        base_exec_prefix = config_vars.get("exec_prefix")
```

Comment:
This one seems to be a bit weird though. exec_prefix and base_exec_prefix are not the same. Base is that of the original interpreter, whereas non-base is that of the active environment.

Reply:
Ah, the `sysconfig._CONFIG_VARS` hold the values of the *original* interpreter.

You may observe this with
```shell
which python3.5
/nix/store/6clz676g768xrshsrwp677nm3n49wsky-python3-3.5.9-env/bin/python3.5

python3.5 -c 'from sysconfig import _CONFIG_VARS; from pprint import pprint; pprint(_CONFIG_VARS)' | grep -i prefix
 'CONFIG_ARGS': "'--prefix=/nix/store/0df1pj661i4m292b0v82gzp0d0i8m6hf-python3-3.5.9' "
 'PYTHONFRAMEWORKPREFIX': '',
 'exec_prefix': '/nix/store/0df1pj661i4m292b0v82gzp0d0i8m6hf-python3-3.5.9',
 'prefix': '/nix/store/0df1pj661i4m292b0v82gzp0d0i8m6hf-python3-3.5.9',
```